src/components/PuzzleView.tsx

```diff
@@ -188,7 +198,19 @@ export const PuzzleView = (props: Props): JSX.Element => {
         puzzleWidth={props.puzzleDefinition.width}
         entryDirection={entryDirection}
         selectedCell={selectedCell}
-        onSelectCell={updateSelectedCell}
+        onSelectCell={(newSelectedCell) => {
+          console.log("onSelectCell", selectedCell, newSelectedCell);
+          if (
+            selectedCell &&
+            newSelectedCell &&
+            newSelectedCell.row === selectedCell.row &&
+            newSelectedCell.column === selectedCell.column
```

Comment:
can you use `?.` safely here? could make this cleaner.

Reply:
probably?